components/StarkNet/modules/getting_started/pages/environment_setup.adoc

```diff
@@ -1,160 +1,208 @@
 
-We recommend working inside a Python virtual environment, but you can also install the `cairo`
-package directly. To create and enter the virtual environment, type:
+* xref:#_option_1__activate_your_cairo_virtual_environment_manually[Activate your Cairo virtual environment manually]
+* xref:#_option_2__use_the_python_environment_manager[Use the Python environment manager]
+
+
+### Option 1 - Activate your Cairo virtual environment manually
+
+Run the following two commands to manually create and activate a Cairo virtual environment:
 
 [,Bash]
 ----
-#Activate your cairo virtual environment
 python3.9 -m venv ~/cairo_venv
 source ~/cairo_venv/bin/activate
 ----
 
-Alternatively, you can also use the `pyenv` environment manager to create a virtual environment. To do so, run the following commands:
+After creating and activating the virtual environment, you should see `(cairo_venv)` in the command line prompt, indicating that the virtual environment is active, for example:
+
+[,Bash]
+----
+(cairo_venv)
+----
+
+
+### Option 2 - Use the Python environment manager
+
+Alternatively, you can use the Python environment manager to create a virtual environment. Simply run the following commands:
 
 [,Bash]
 ----
-#Install pyenv
 curl https://pyenv.run | bash
+----
+
+Now add the following to your `.bashrc` or `.zshrc` file:
 
-#Add the following to your .bashrc or .zshrc
+[,Bash]
+----
 export PATH="$HOME/.pyenv/bin:$PATH"
 eval "$(pyenv init -)"
 eval "$(pyenv virtualenv-init -)"
+----
+
+Restart your terminal and run the following commands:
 
-#Restart your terminal and run the following commands
+[,Bash]
+----
 pyenv install 3.9.0
 pyenv virtualenv 3.9.0 cairo_venv
 pyenv activate cairo_venv
 ----
 
-Either way, make sure the `venv` is activated – you should see (`cairo_venv`) in the command line prompt.
-
-Your terminal should look like this:
+you should see `(cairo_venv)` in the command line prompt, indicating that the virtual environment
+is active, for example:
 
 [,Bash]
 ----
-(cairo_venv) satoshi@nakamoto-notebook % 
-
+(cairo_venv)
 ----
 
-=== Step 2: Install the necessary dependencies
+## Install the additional dependencies
+
+Depending on your operating system, you may need to install the following, additional dependencies:
+|===
+|Dependency |Description
+
+|`libgmp3-dev`/`gmp`|A software development library package for the GNU Multiple Precision Arithmetic
+Library.
+|`ECDSA` |(Elliptic Curve Digital Signature Algorithm) is a digital signature algorithm that is based on the mathematics of elliptic curves.
+|`FastECDSA`|An open-source software library that provides an implementation of the Elliptic Curve
+Digital Signature Algorithm (ECDSA) for use in cryptographic applications. It is designed to be fast, efficient, and easy to use, and is written in the Python programming language.
+|`SymPy` |A Python library for symbolic mathematics.
+|===
 
-On Ubuntu, for example, you will have to first run:
+
+### On Linux Ubuntu
```

Comment:
```suggestion
=== Linux
```
You should state what the OS prerequisites are as well. If this has only been tested on a particular version of an OS, then state that, perhaps indicating a minimum version. Is only Ubuntu supported? If not, then you should state how to install on the major distros, such as Red Hat and Suse, or at the very least state that your example uses Ubuntu, but if using a different distro, use that appropriate package manager for that distro.

Also if Windows users are supposed to use a Linux shell, then you should state that as well.